streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -204,14 +206,40 @@ public TopologyTestDriver(final Topology topology,
     public TopologyTestDriver(final Topology topology,
                               final Properties config,
                               final long initialWallClockTimeMs) {
+
+        this(topology.internalTopologyBuilder, config, initialWallClockTimeMs);
+    }
+
+    /**
+     * Create a new test diver instance.
+     *
+     * @param builder builder for the topology to be tested
+     * @param config the configuration for the topology
+     */
+    public TopologyTestDriver(final InternalTopologyBuilder builder,
```

Comment:
This class is part of public API -- thus, we cannot change any public interface without a KIP. I am also not sure, if we should change this class in the first place. What limitations did you find rewriting the tests? We should try to rewrite out test without any change to this class IMHO.

For this particular change, we should not expose `InternalTopologyBuilder` because it's an internal class.

Reply:
Alternatively, we can avoid using this driver for that test, or to rewrite the test to give a `Topology` instead of an `InternalTopologyBuilder`.

IMO, the function of this class should be strictly limited to what makes sense for users to test their topologies. If we bolt stuff on for testing internals, it paves the way to pollute this class's purpose.

I'm all for using this class for internal tests, but if we need something that users wouldn't need, then we should use an internal driver instead of adding backdoors for internal testing to this class.

Reply:
I followed @mjsax 's suggestion and implemented a wrapper class which exposes the protected constructor. It is only used in a few places, and the tests where it is used are for deprecated classes should be removed in the future. This way we can still use `TopologyTestDriver` for these internal tests, while keeping the public interface unchanged.

But I agree with @vvcephei . While the Jira ticket asks for a removal of the KStreamsTestDriver class, I think it is wise to keep it around for those cases where we need the added functionality, to avoid polluting the `TopologyTestDriver`.